src/handler.js

```diff
@@ -0,0 +1,330 @@
+      }
+
+      if (treeherderConfig.groupName) {
+        job.display.groupName = treeherderConfig.groupName;
+      }
+
+      if (task.extra.chunks) {
+        job.display.chunkCount = task.extra.chunks.total;
+        job.display.chunkId = task.extra.chunks.current;
+      }
+    } catch(err) {
+      throw new Error(`Error constructing pulse message. ${err}, ${err.stack}`);
+    }
+
+    return job;
+  }
+
+  async handleTaskPending(pushInfo, task, message) {
+    let job = this.buildJobMessage(pushInfo,
+                             message.status.taskId,
+                             message.runId,
+                             task,
+                             message.status.runs[message.runId]);
+    await this.publishJobMessage(pushInfo, job);
+  }
+
+  async handleTaskRerun(pushInfo, task, message) {
+    let run = message.status.runs[message.runId-1];
+    let job = this.buildJobMessage(pushInfo,
+                             message.status.taskId,
+                             message.runId-1,
+                             task,
+                             run);
+    job.state = 'completed';
+    job.result = 'fail';
+    job.isRetried = true;
+    job.logs = [createLogReference(this.queue, message.status.taskId, run)];
+    await this.publishJobMessage(pushInfo, job);
+  }
+
+  async handleTaskRunning(pushInfo, task, message) {
+    let run = message.status.runs[message.runId];
+    let job = this.buildJobMessage(pushInfo,
+                             message.status.taskId,
+                             message.runId,
+                             task,
+                             run);
+    job.timeStarted = run.started;
+    job.logs = [createLogReference(this.queue, message.status.taskId, run)];
+    await this.publishJobMessage(pushInfo, job);
+  }
+
+  async handleTaskCompleted(pushInfo, task, message) {
+    let run = message.status.runs[message.runId];
+    let job = this.buildJobMessage(pushInfo,
+                             message.status.taskId,
+                             message.runId,
+                             task,
+                             run);
+
+    job.timeStarted = run.started;
+    job.timeCompleted = run.resolved;
+    job.logs = [createLogReference(this.queue, message.status.taskId, run)];
+    await this.publishJobMessage(pushInfo, job);
+  }
+
+  async handleTaskFailed(pushInfo, task, message) {
+    // To correctly handle the rerun case we must not mark jobs which will be
+    // marked as retry as 'completed'. This means we must determine if this run
+    // will trigger a retry by querying the scheduler.
+    if (
+      task.schedulerId === 'task-graph-scheduler' &&
+      task.taskGroupId
+    ) {
+      let taskInfo = await this.scheduler.inspectTask(task.taskGroupId, message.status.taskId);
+      if (taskInfo.reruns > message.runId) {
+        // Simply allow the rerun handle to update the task...
+        return;
+      }
+    }
```

Comment:
like we don't want to keep this, agree? when scheduler goes away?

What if we just don't keep it?